src/Features/Core/Portable/InlineHints/AbstractInlineParameterNameHintsService.cs

```diff
@@ -273,19 +273,28 @@ static bool SuffixMatchesParameterName(ReadOnlyMemory<char> suffix, string param
         }
     }
 
-    private static bool ParameterMatchesArgumentName(string? identifierArgument, IParameterSymbol parameter, ISyntaxFactsService syntaxFacts)
-    => syntaxFacts.StringComparer.Compare(parameter.Name, identifierArgument) == 0;
+    private static bool ParameterMatchesArgumentName(SyntaxNode argument, IParameterSymbol parameter, ISyntaxFactsService syntaxFacts)
+    {
+        var argumentName = GetIdentifierNameFromArgument(argument, syntaxFacts);
+        return syntaxFacts.StringComparer.Compare(parameter.Name, argumentName) == 0;
+    }
 
     protected static string GetIdentifierNameFromArgument(SyntaxNode argument, ISyntaxFactsService syntaxFacts)
     {
         var identifierNameSyntax =
             syntaxFacts.IsArgument(argument) ? syntaxFacts.GetExpressionOfArgument(argument) :
             syntaxFacts.IsAttributeArgument(argument) ? syntaxFacts.GetExpressionOfAttributeArgument(argument) : null;
 
+        if (syntaxFacts.IsMemberAccessExpression(identifierNameSyntax))
+        {
+            identifierNameSyntax = syntaxFacts.GetNameOfMemberAccessExpression(identifierNameSyntax);
+        }
+
         if (!syntaxFacts.IsIdentifierName(identifierNameSyntax))
             return string.Empty;
 
         var identifier = syntaxFacts.GetIdentifierOfIdentifierName(identifierNameSyntax);
```

Comment:
I believe there is a TrGetInferredMemberName you could potentially use. Up to you.